### PR TITLE
Implements std::error::Error and std::fmt::Display for Error

### PR DESCRIPTION
**Pull request by bkchr**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -91,6 +91,31 @@ pub enum Error {
...
+        match *self {
+            Error::Io(ref io) => io.description(),
+            Error::Serde(ref json) => json.description(),
+        }
+    }
+
+    fn cause(&self) -> Option<&error::Error> {
+        match *self {
+            Error::Io(ref io) => io.cause(),
+            Error::Serde(ref serde) => serde.cause(),
```
Comment (dtolnay) [inline src/lib.rs]: The cause of the current error is `io` or `serde`.

Comment (bkchr) [reply src/lib.rs]: Sorry, I don't understand what you want to say to me ^^

Comment (dtolnay) [reply src/lib.rs]: If I have an error of type tokio_serde_json::Error, the immediate cause of that error is either an error of type io::Error or an error of type serde_json::Error. So the implementation of `cause` needs to return `io` or `serde`. As currently written, it does not return `io` or `serde`. It returns whatever caused one of those.

Comment (bkchr) [reply src/lib.rs]: Ahh yeah, good catch! Now it should be okay :)

**Pull request merged**